src/smt/theory_str_mc.cpp

```diff
@@ -770,7 +771,7 @@ namespace smt {
                 // convert iValue to a constant
                 zstring iValue_str(iValue.to_string());
                 for (unsigned idx = 0; idx < iValue_str.length(); ++idx) {
-                    expr_ref chTerm(bitvector_character_constants.get(iValue_str[idx]), sub_m);
+                    expr_ref chTerm(u.mk_char(iValue_str[idx]), sub_m);
```

Comment:
sub_m has to be the same as m, otherwise this will crash. u.mk_char refers to m.
So suggestion is to get rid of sub_m.